modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -28,11 +44,59 @@ export interface UploadResult {
 }
 
 interface EditorUpload {
+  /**
+   * Cache of blob elements created in an editor instance.
+   *
+   * @property blobCache
+   * @type Object
+   */
   blobCache: BlobCache;
+
+  /**
+   * Adds a custom filter to be executed before any images are uploaded to the server.
+   * Images must return true on every added filter to be uploaded, else they are filtered out.
```

Comment:
```suggestion
   * Adds a custom filter that is executed before any images are uploaded to the server.
   * Images must return true on every added filter to be uploaded.
   * An image that returns false on any added filter is filtered out (ie, not uploaded).
```

1. Switched from the quasi–future-tense infinitive — _to be_ — to the present tense indicative — _is_.
2. Went all in on setting out what happens on returning either *true* and *false*.

NB: As it stands, both the original and edited text infer something that, if true, is probably worth making absolutely explicit.

They both infer that, if you have x images to upload, no uploading of any images begins until all x images run through all added filters.

For large values of x, I’m guessing that could result in a UX worth being explicit about.

If this is the case, I’d argue for something like the following:

```suggestion
   * Adds a custom filter that is executed before any images are uploaded to the server.
   * Images must return true on every added filter to be uploaded.
   * An image that returns false on any added filter is filtered out (ie, not uploaded).
   * NB: no images are uploaded until all added filters are executed over all supplied images.
```

OTOH, if the process is

1. run filters over image-x.
2. if true upload image-x.
3. if false do not upload image-x.
4. run filters over image-x+1
5. if true, upload image-x
6. if false, do not upload image-x+1.
7. repeat until image-x+n (where x+n = last image in the list).

that would be a different UX. And, if this latter is the case, I’d argue for something like the following:

```suggestion
   * Adds a custom filter that is executed before an image is uploaded to the server.
   * Images must return true on every added filter to be uploaded.
   * An image that returns false on any added filter is filtered out (ie, not uploaded).
   * NB: individual images are uploaded as soon as they return true to every added filter.
```

Reply:
Apologies for the confusion, the filters are applied during the scanForImages method, which is called from uploadImages. I'll fix the wording now